### DISCUSSION TITLE
`back-edge from insn X to Y` on kernel 4.19

### DISCUSSION THREAD

**Discussion by xiaonuoz** [cilium/ebpf#848] [Q&A] [answered]
Hi there,
The kernel of the development environment is 5.10.0,It runs without any problems.
The kernel of a system that failed to execute the program is 4.19.0.It doesn't have a /sys/kernel/btf folder.
![image](https://user-images.githubusercontent.com/46923935/200468113-41c208f6-f960-45fd-ad79-725abd7c7c3d.png)

so,I changed the address of LoadKernelSpec() to the current project and downloaded a corresponding kernel of vmlinux.
![image](https://user-images.githubusercontent.com/46923935/200468778-24ab2706-c3fb-411b-83ce-9b4d5f9460b4.png)
![image](https://user-images.githubusercontent.com/46923935/200468837-685ca083-6593-47e4-ad57-166df80b7bc7.png)

I don't know if this is the right migration.It sent an error.
![image](https://user-images.githubusercontent.com/46923935/200467976-cc180bfb-f38b-48d6-b0cc-d76d34227542.png)

Can you help me?
Thanks!

**Comment by ti-mo** [comment, marked as answer]
`back-edge from insn 82 to 57` means your BPF program contains a loop (a backward jump). This has only been supported as of kernel 5.3, see: https://lwn.net/Articles/794934.
